web/html/src/components/dialog/ReactDialog.tsx

```diff
@@ -0,0 +1,64 @@
+import * as React from "react";
+import ReactModal from 'react-modal';
+import "./Dialog.css";
+
+export type DialogProps = {
+  /** whether the dialog should be shown or hidden */
+  isOpen: boolean;
+  /** Function to call to close the modal. This typically changes the value of isOpen */
+  onClose?: () => void;
+  /** The id of the html div tag */
+  id: string;
+  /** The css className for the 'modal-dialog' div */
+  className?: string;
+  /** Title of the dialog */
+  title?: React.ReactNode;
+  /** The body of the popup */
+  content?: React.ReactNode;
+  /** The footer content of the dialog. Usually buttons. */
+  footer?: React.ReactNode;
+  /** Should the dialog show a header with a title. Defaults to False */
+  hideHeader?: boolean;
+  /** Can the modal be closed using the cross button, ESC or by clicking on the overlay? */
+  closableModal?: boolean;
+}
+
+export function Dialog(props: DialogProps) {
+  const closableModal = props.closableModal ?? true;
+
+  ReactModal.setAppElement(document.body);
```

Comment:
This looks like it might belong outside of this function call, unless it's intentional to call this a number of times? But I might be wrong here, I haven't checked the API docs for ReactModal.

Reply:
indeed, that should only been called once before any modal dialog is shown. Where should that be done? Would it work right above the component function?

Reply:
Yes, in that case it will get called only once.